internal/bmcutils/bmcutils.go

```diff
@@ -175,3 +177,57 @@ func CreateBMCClient(
 func GetServerNameFromBMCandIndex(index int, bmc *metalv1alpha1.BMC) string {
 	return fmt.Sprintf("%s-%s-%d", bmc.Name, "system", index)
 }
+
+func SSHResetBMC(ctx context.Context, ip, manufacturer, username, password string, timeout time.Duration) error {
+	// If Redfish reset fails, try SSH-based reset for known manufacturers
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{ssh.Password(password)},
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+			return nil
+		},
+		Timeout: timeout,
+	}
+	client, err := ssh.Dial("tcp", net.JoinHostPort(ip, "22"), config)
+	if err != nil {
+		return fmt.Errorf("failed to dial ssh: %w", err)
+	}
+	defer func() {
+		_ = client.Close()
+	}()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create ssh session: %w", err)
+	}
+	defer func() {
+		_ = session.Close()
+	}()
+	resetCMD := ""
+	switch manufacturer {
```

Comment:
I would check for manufacturer support in the very beginning of the function (before creating any connections).